scipy/stats/_distn_infrastructure.py

```diff
@@ -1426,8 +1426,26 @@ def support(self, *args, **kwargs):
 
         """
         args, loc, scale = self._parse_args(*args, **kwargs)
-        _a, _b = self._get_support(*args)
-        return _a * scale + loc, _b * scale + loc
```

Comment:
Did you consider changing just changing the last line to:
```python3
out_a, out_b = _a * scale + loc, _b * scale + loc
cond = self._argcheck(*args) & (scale > 0)
place(out_a, ~cond, self.badvalue)
place(out_b, ~cond, self.badvalue)
if loc.ndim == 0:
   return out_a.item(), out_b.item()
return out_a, out_b
```
I wonder if it is worth all the trouble to split it all apart and put it back together. Are any of the support calculations time consuming? Do any of them cause errors with invalid parameters? 

Alternatively, would `_lazywhere` simplify things?

Reply:
> Are any of the support calculations time consuming?

I don't think some floating-point operations will be too time-consuming.

> Do any of them cause errors with invalid parameters?

No, I don't think so.

> would `_lazywhere` simplify things?

It also seems like a nice fix. I will try to implement one of these alternatives and push it. Thanks for noticing this!